### PR TITLE
2671: Refactoring of ding_ipe_filter

### PR DESCRIPTION
**Pull request by cableman**
* New administration interface
* New filter logic

See https://platform.dandigbib.org/issues/2671

### CODE REVIEW

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 7b6031eea1ac3a40db05e452de5ae4fbfbcad9d7
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 7b6031eea1ac3a40db05e452de5ae4fbfbcad9d7
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 7b6031eea1ac3a40db05e452de5ae4fbfbcad9d7
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 7b6031eea1ac3a40db05e452de5ae4fbfbcad9d7
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by cableman** [comment]
The scrutinizer error is about the use of a pane title given to t(), but right now I don't see another solution.

File: modules/ding_ipe_filter/ding_ipe_filter.module [hunk 1]
```diff
@@ -191,3 +205,33 @@ function _ding_ipe_filter_show_for_admin() {
...
+    'ding-:all_opening_hours' => 'ding-:all_opening_hours',
+    'ding-:popular' => 'ding-:popular',
+    'ding-:interaction_pane' => 'ding-:interaction_pane',
+    'ding-:serendipity_ting_object' => 'ding-:serendipity_ting_object',
+    'ding-:ding_tabroll-ding_frontpage_tabroll' => 'ding-:ding_tabroll-ding_frontpage_tabroll',
+    'event-panes:ding_event-ding_event_list_frontpage' => 'event-panes:ding_event-ding_event_list_frontpage',
+    'event-panes:ding_event-ding_event_simple_list' => 'event-panes:ding_event-ding_event_simple_list',
+    'groups-panes:ding_groups-ding_group_overview_simple' => 'groups-panes:ding_groups-ding_group_overview_simple',
+    'news-panes:ding_news-ding_news_frontpage_list' => 'news-panes:ding_news-ding_news_frontpage_list',
+    'ting:carousel' => 'ting:carousel',
```
Comment (kasperg) [inline modules/ding_ipe_filter/ding_ipe_filter.module]: Is there any reason to have the same values as both key and value? It seems very verbose. I would expect a boolean value to be sufficient for our usecase e.g. 

```php
$default = array(
  'ding-:all_opening_hours' => TRUE,
  'ding-:popular' => TRUE,
// and so on

File: modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        'category' => $category['title'],
+        'type' => $content['subtype_name'],
+        'title' => $content['title'],
+        'description' => isset($content['description']) ? check_plain($content['description']) : '',
+      );
+    }
+  }
+
+  $form['ding_ipe_filter_panes']['ding_ipe_filter_panes_selected'] = array(
+    '#type' => 'tableselect',
```
Comment (kasperg) [inline modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc]: :+1: This seems like a very nice way to utilize existing form API elements to create a administrative interface.

File: modules/ding_ipe_filter/ding_ipe_filter.module [hunk 3]
```diff
@@ -122,56 +90,102 @@ function ding_ipe_filter_theme_registry_alter(&$registry = array()) {
...
   $region_id  = $variables['region_id'];
-  $renderer = $variables['renderer'];
 
-  $buttons = array();
-  $attributes = array(
-    'class' => 'ctools-use-modal ding-ipe-menu-item',
+  $output = array(
+    '#theme' => 'ding_ipe_filter_selection',
+    '#region_id' => $region_id,
+    '#links' => $link = array(
```
[the remark this answers is not in the document]
Comment (cableman) [reply modules/ding_ipe_filter/ding_ipe_filter.module]: Copy & past error, fixed

File: modules/ding_ipe_filter/ding_ipe_filter.module [hunk 4]
```diff
@@ -191,3 +205,33 @@ function _ding_ipe_filter_show_for_admin() {
...
 }
+
+/**
+ * Get the current filter list.
+ *
+ * The panel panes selected in the administration UI and if not configured yet
+ * the default selections is used.
+ *
+ * @param bool $filter
+ *   If TRUE the field with value else all filters.
```
[the remark this answers is not in the document]
Comment (cableman) [reply modules/ding_ipe_filter/ding_ipe_filter.module]: Removed as it's not needed.

File: modules/ding_ipe_filter/ding_ipe_filter.module (hunk 1 above)
Comment (cableman) [reply modules/ding_ipe_filter/ding_ipe_filter.module]: It's the way drupal forms save the value in the database first time you save them... so...

File: modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  $form['ding_ipe_filter_settings']['admin_display'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Display IPE menu for super admin'),
+    '#default_value' => !empty($configs['admin_display']) ? $configs['admin_display'] : 0,
+  );
+
+  $form['ding_ipe_filter_panes'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Menu items'),
+    '#description' => t('Select the panel panes that should be available in the editor when clicking the "add" buttons.'),
```
[the remark this answers is not in the document]
Comment (cableman) [reply modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc]: It's form the original module (it's the item available in the pop-up menu), but renamed now.

File: modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  // Select table header definition.
+  $header = array(
+    'category' => t('Category'),
+    'type' => t('Type'),
+    'title' => t('Title'),
+    'description' => t('Description'),
+  );
+
+  // Get all panes based on there categories.
```
[the remark this answers is not in the document]
Comment (cableman) [reply modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc]: Fixed

File: modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc (hunk 5 above)
[the remark this answers is not in the document]
Comment (cableman) [reply modules/ding_ipe_filter/includes/ding_ipe_filter.admin.inc]: :-)

File: modules/ding_ipe_filter/ding_ipe_filter.module (hunk 1 above)
Comment (cableman) [reply modules/ding_ipe_filter/ding_ipe_filter.module]: :-)

**Pull request merged**